File: 2337-remove-digit-from-number-to-maximize-result/remove-digit-from-number-to-maximize-result.py
class Solution:
    def removeDigit(self, number: str, digit: str) -> str:
        n = len(number)
        current_num = int(number)
        max_number = 0
        for i, num in enumerate(number):
            if num == digit:
                new_number = number[:i] + number[i+1:]
                if max_number < int(new_number):
                    max_number = int(new_number)
        return str(max_number)

File: 2337-remove-digit-from-number-to-maximize-result/remove-digit-from-number-to-maximize-result.java
class Solution {
    public String removeDigit(String number, char digit) {
        String ans = "";
        String temp = "";
        for (int i = 0; i < number.length(); ++i){
            if (number.charAt(i) == digit){
                temp = number.substring(0, i) + number.substring(i+1);
                if (temp.compareTo(ans) > 0) {
                    ans = temp;
                }
            }
        }
        return ans;
    }
}